discovery/gossiper.go

```diff
@@ -2454,23 +2454,10 @@ func (d *AuthenticatedGossiper) updateChannel(info *channeldb.ChannelEdgeInfo,
 	}
 	edge.LastUpdate = time.Unix(timestamp, 0)
 
-	chanUpdate := &lnwire.ChannelUpdate{
-		ChainHash:       info.ChainHash,
-		ShortChannelID:  lnwire.NewShortChanIDFromInt(edge.ChannelID),
-		Timestamp:       uint32(timestamp),
-		MessageFlags:    edge.MessageFlags,
-		ChannelFlags:    edge.ChannelFlags,
-		TimeLockDelta:   edge.TimeLockDelta,
-		HtlcMinimumMsat: edge.MinHTLC,
-		HtlcMaximumMsat: edge.MaxHTLC,
-		BaseFee:         uint32(edge.FeeBaseMSat),
-		FeeRate:         uint32(edge.FeeProportionalMillionths),
-		ExtraOpaqueData: edge.ExtraOpaqueData,
-	}
-
 	// With the update applied, we'll generate a new signature over a
 	// digest of the channel announcement itself.
-	sig, err := netann.SignAnnouncement(
+	chanUpdate := netann.UnsignedChannelUpdateFromEdge(info, edge)
```

Comment:
do we need the `unsigned` version? can't we call  `ChannelUpdateFromEdge` here directly, and then reset the sig?

Reply:
fixed, we also needed to update the timestamp as well

Reply:
let me know if the new solution is acceptable. the issue was that the sig is empty and then `ChannelUpdateFromEdge` fails. i've changed `ChannelUpdateFromEdge` to return both the update and error so that the call site can just ignore. I have a slight preference for the having the unsigned version though, that way we don't accidentally ignore errors in the future if the method changes.

Reply:
Hm, yeah this looks a bit brittle. Maybe a simple unsigned version is preferred.

Reply:
Did you decide on anything here? Possibly the signing/sigparsing can me made a modifier as well, such that we can avoid that brittle error handling

Reply:
reverted to use `UnsignedChannelUpdateFromEdge` bc i think it's the most explicit